### PR TITLE
[PACT-865] Now retrieving 'places of deposit' via message bus.

### PR DESCRIPTION
**Pull request by tna-erasmos**

### CODE REVIEW

**Comment by tna-erasmos** [comment]
One of my guiding constraints in this was to only retrieve the places of deposit when used.

**Review by felixgb — APPROVED** [review]

File: app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala [hunk 1]
```diff
@@ -96,45 +95,49 @@ class EditSetRecordController @Inject() (
...
-        BadRequest(generateEditSetRecordEditView(user, editSet, record, formWithErrors))
+        referenceDataService.getPlacesOfDeposit().map { placesOfDeposit =>
+          BadRequest(generateEditSetRecordEditView(user, editSet, record, placesOfDeposit, formWithErrors))
+        }
       case Left(EditSetNotFound(missingEditSetId)) =>
         BadRequest(s"Edit Set with ID [$missingEditSetId] not found")
       case Left(EditSetRecordNotFound(missingOci)) => BadRequest(s"Record with $missingOci not found")
       case Left(InvalidAction(badAction))          => BadRequest(s"$badAction is not allowed action")
       case Left(MissingAction)                     => BadRequest("This action is not allowed")
+      case Left(anotherError) => InternalServerError(s"Error occurred: [$anotherError]") // TODO: Untested
```
Comment (userjd) [inline app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala:131]: do we need a catch all here?

File: app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala [hunk 2]
```diff
@@ -172,117 +175,136 @@ class EditSetRecordController @Inject() (
...
+  )(implicit
     request: Request[AnyContent]
-  ): Result = {
-    val selectedNonEmptyCreatorsFromRequest = filterRequestData { case (key, value) =>
-      key.startsWith(FieldNames.creatorIDs) && value.trim.nonEmpty
+  ): IO[Result] =
+    referenceDataService.getPlacesOfDeposit().map { placesOfDeposit =>
+      val selectedNonEmptyCreatorsFromRequest = filterRequestData { case (key, value) =>
+        key.startsWith(FieldNames.creatorIDs) && value.trim.nonEmpty
+      }
```
Comment (userjd) [inline app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala:235]: Should the map be for the entire code block?

Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala:235]: Correct.

I'll change it.

Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala:235]: Actually, it is already: I was mistaken.

File: app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala (hunk 1 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala]: OK

File: app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala (hunk 2 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala:234]: OK

**Review by userjd — APPROVED** [review]

**Review by rwalpole — APPROVED** [review]

**Pull request merged**